File: sample-jdbc-java/src/main/java/com/ucsoftworks/leafdb/jdbc/Sample.java
package com.ucsoftworks.leafdb.jdbc;

import com.ucsoftworks.leafdb.LeafDb;
import com.ucsoftworks.leafdb.dsl.Condition;
import com.ucsoftworks.leafdb.dsl.Field;
import com.ucsoftworks.leafdb.dsl.FieldExtensionsKt;
import kotlin.jvm.functions.Function1;

import java.util.LinkedList;
import java.util.List;

public class Sample {
    private static final String TABLE_TEST = "test";
    private static final Field A = new Field("a"), B = new Field("b");

    public static void main(String[] args) {
        final SampleHelper sampleHelper = new SampleHelper("sample-java.db", 1);
        final LeafDb leafDb = sampleHelper.getLeafDb();

        leafDb.insertOrUpdate(TABLE_TEST, new SampleData(1, 2), A).execute();
        leafDb.insertOrUpdate(TABLE_TEST, new SampleData(3, 4), A).execute();
        leafDb.insertOrUpdate(TABLE_TEST, new SampleData(5, 6), A).execute();

        final Field f = FieldExtensionsKt.minus(5, A);

        leafDb.updateAll(TABLE_TEST, new LinkedList<SampleData>() {{
            add(new SampleData(3, 3));
        }}, new Function1<SampleData, Condition>() {
            @Override
            public Condition invoke(SampleData sampleData) {
                return A.equal(sampleData.a);
            }
        }).execute();

        final List<SampleData> dbData = leafDb.select(TABLE_TEST).where(A.plus(B).greaterThan(5)).findAs(SampleData.class).execute();

        System.out.println(dbData);
    }

    private static class SampleData {
        int a;
        int b;

        SampleData(int a, int b) {
            this.a = a;
            this.b = b;

        }

        @Override
        public String toString() {
            return "SampleData{" +
                    "a=" + a +
                    ", b=" + b +
                    '}';
        }
    }

}

File: sample-jdbc/src/main/java/com/ucsoftworks/leafdb/Sample.kt
package com.ucsoftworks.leafdb

import com.ucsoftworks.leafdb.dsl.Field
import java.util.*

/**
 * Created by Pasenchuk Victor on 28/08/2017
 */

data class S(val i: Int, val s: String)

data class S1(val i: Int, val s: String, val t: Int?)

val I = Field("i")
val J = Field("j")

fun main(args: Array<String>) {

    val leafDbJdbcHelper = SampleHelper("sample.db", 3)

    val leafDb = leafDbJdbcHelper.leafDb

    leafDb.insertOrUpdate(Tables.TEST.tableName, S(1, "asd"), I).execute()
    leafDb.insertOrUpdate(Tables.TEST.tableName, S(5, "qwe"), I).execute()

    leafDb.partialUpdate(Tables.TEST.tableName, mapOf("i" to 1, "t" to 4), I).execute()

    leafDb.partialUpdate(Tables.TEST.tableName, mapOf("t" to 4), I equal 5).execute()

    System.out.println(leafDb.engineVersion().execute())

    System.out.println(leafDb.select(Tables.TEST.tableName).findAs(S1::class.java).execute())

    val s = S(1, "1")
    System.out.println(Date())
    for (i in 1..1_000)
        leafDb.insert(Tables.LOAD.tableName, s).execute()

    System.out.println(Date())

}
